### PR TITLE
JSCOE-79 Code an experiment that shows how to build a Full Stack JS WebSocket application

### PR DESCRIPTION
**Pull request by vcnwafor**
- pulled repo under version control
- fixed dependency list
- Clean up on ReadMe files
- did some house keeping before containerization.
- containerized the services
- containerized all the services.
- fixed minor issues before implementing redis.
- fixed issues with haproxy config
- after integrating redis.

### CODE REVIEW

File: fullstack-js-websocket-experiment/backend/chat_withDB.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+            });
+
+            members.splice(index, 1);
+            this.saveMembers(members);
+
+            this.socket.off('join', this.join);
+            this.socket.off('room.join', this.joinRoom);
+            this.socket.off('room.leave', this.leaveRoom);
+            this.socket.off('room.getMessages', this.getRoomMessages);
+            this.socket.off('room.message', this.sendMessageToRoom);
```
Comment (rafaelwaterkemper) [inline fullstack-js-websocket-experiment/backend/chat_withDB.js:120]: You can use the socket.removeAllListeners();

File: fullstack-js-websocket-experiment/backend/chat_withDB.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+            this.socket.disconnect();
```
Comment (rafaelwaterkemper) [inline fullstack-js-websocket-experiment/backend/chat_withDB.js:122]: Calling disconnect again?

Comment (vcnwafor) [reply fullstack-js-websocket-experiment/backend/chat_withDB.js]: thanks, fix applied.

**Comment by rafaelwaterkemper** [comment]
@vcnwafor @rudolf Nice experiment dude! I can just suggest to clean up some files from project that aren't being used like (App.css, App.test and others unused).
Also I'd suggest to organize better the backend project, have so many files inside the root "src" folder.

For backend project, the Dockerfile has the production stage which is calling npm build from CMD, therefore, the package.json doesn't have "build" script.

For frontend project, the Dockerfile has the production stage and "build" script, therefore, we don't have any next step to use the builded files to be "served" like for an Ngnix... It's just for demonstration? Or it'd been prepared for a future branch?

For last, I saw the project have many ways to build the server using clusters, pm2, with db (node-storage) and using redis. I'd suggest to organize these files in a proper way and also, if is relevant, add into a README file when and why using each option.

Overall, the recommendations above are to improve the organization of the repo and facilitate the understanding of the project by another readers.

**Pull request merged**